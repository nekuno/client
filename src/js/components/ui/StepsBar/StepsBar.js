import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './StepsBar.scss';

export default class StepsBar extends Component {

    static propTypes = {
        color         : PropTypes.oneOf(['purple', 'blue', 'pink', 'green']),
        totalSteps    : PropTypes.number,
        currentStep   : PropTypes.number,
        canContinue   : PropTypes.bool,
        continueText: PropTypes.string,
        cantContinueText: PropTypes.string,
        onClickHandler: PropTypes.func
    };

    handleClick() {
        const {canContinue} = this.props;
        if (canContinue && this.props.onClickHandler) {
            this.props.onClickHandler();
        }
    }

    renderDots = function(totalSteps, currentStep) {
        let dots = [];

        for (let i = 0; i < totalSteps; i++) {
            dots.push(<li key={i} className={i === currentStep ? styles.active : null}>
                <div className={styles.dot}/>
            </li>);
        }

        return <ul className={styles.dots}>
            {dots.map(dot => dot)}
        </ul>;
    };

    render() {
        const {color, totalSteps, currentStep, canContinue, continueText, cantContinueText} = this.props;
        const textClass = totalSteps > 0 ?
            canContinue ? styles.continue : styles.cantContinue
            : canContinue ? styles.noSteps + ' ' + styles.continue : styles.noSteps + ' ' + styles.cantContinue;

        let stepsBarClass = styles.stepsBar + ' ' + styles[color];
        stepsBarClass = !canContinue ? stepsBarClass + ' ' + styles.cantContinue : stepsBarClass;

        return (
            <div className={styles.stepsBarWrapper}>
                <div className={stepsBarClass} onClick={this.handleClick.bind(this)}>
                    {totalSteps > 0 ?
                        this.renderDots(totalSteps, currentStep)
                        : null
                    }
                    <div className={textClass + ' small'}>
                        {canContinue ? continueText + ' ' : cantContinueText}
                        {totalSteps > 0 && canContinue ? <span className="icon icon-arrow-right" /> : null}
                    </div>
                </div>
            </div>
        );
    }
}