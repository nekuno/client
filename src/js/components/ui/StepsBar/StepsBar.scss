@import '../../../../scss/_partials/colors';

.steps-bar-wrapper {

  .steps-bar {
    background: $purple-main;
    box-sizing: border-box;
    color: $white-background;
    cursor: pointer;
    font-weight: 600;
    height: 48px;
    position: absolute;
    bottom: 0;
    width: 100%;
    z-index: 11111;

    .dots {
      display: inline-block;
      list-style: none;
      margin: 0;
      height: 48px;
      padding-left: 5%;
      text-align: left;
      vertical-align: bottom;
      width: 45%;

      li {
        display: inline-block;
        height: 48px;
        line-height: 48px;
        margin: 0;

        .dot {
          background-color: #fff;
          border: 8px solid rgba(255, 255, 255, 0);
          background-clip: padding-box;
          border-radius: 50%;
          display: inline-block;
          height: 4px;
          width: 4px;
        }
        &.active {
          .dot {
            border: 8px solid rgba(255, 255, 255, 0.2);
          }
        }
      }
    }

    .continue, .cant-continue {
      display: inline-block;
      line-height: 48px;
      width: 40%;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      padding: 0 5%;
      position: relative;
      -webkit-box-flex: 1;
      -webkit-flex-shrink: 1;
      -ms-flex: 0 1 auto;
      flex-shrink: 1;
      min-width: 0;
    }
    .continue {
      text-align: right;
    }
    .cant-continue {
      font-weight: normal;
      text-align: center;
    }

    .no-steps {
      text-align: center;
      width: 90%;
    }

    &.blue {
      background: $blue-projects;

      &.cant-continue {
        .dots {
          li {
            .dot {
              background-color: $blue-projects;
              border: 8px solid rgba(255, 255, 255, 0);
            }
            &.active {
              .dot {
                border: 8px solid $blue-projects-translucent;
              }
            }
          }
        }
      }
    }
    &.pink {
      background: $pink-hobbies;

      &.cant-continue {
        .dots {
          li {
            .dot {
              background-color: $pink-hobbies;
              border: 8px solid rgba(255, 255, 255, 0);
            }
            &.active {
              .dot {
                border: 8px solid $pink-hobbies-translucent;
              }
            }
          }
        }
      }
    }
    &.green {
      background: $green-experiences;

      &.cant-continue {
        .dots {
          li {
            .dot {
              background-color: $green-experiences;
              border: 8px solid rgba(255, 255, 255, 0);
            }
            &.active {
              .dot {
                border: 8px solid $green-experiences-translucent;
              }
            }
          }
        }
      }
    }

    &.cant-continue {
      background: $white-background;
      color: $gray-secondary-text;
      cursor: default;

      .dots {
        li {
          .dot {
            background-color: $purple-main;
            border: 8px solid rgba(255, 255, 255, 0);
          }
          &.active {
            .dot {
              border: 8px solid $purple-main-translucent;
            }
          }
        }
      }
    }
  }
}
