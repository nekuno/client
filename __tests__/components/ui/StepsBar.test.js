import React from 'react';
import StepsBar from '../../../src/js/components/ui/StepsBar/StepsBar.js';
import styles from '../../../src/js/components/ui/StepsBar/StepsBar.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test StepsBar component', () => {
    const mockFn = jest.fn();
    const stepsBar = shallow(<StepsBar totalSteps={3} currentStep={0} continueText={'Continue'} cantContinueText={'You cannot continue'} canContinue={true} onClickHandler={mockFn}/>);

    it('should be defined', () => {
        expect(StepsBar).toBeDefined();
    });
    it('should render correctly', () => {
        expect(stepsBar).toMatchSnapshot();
    });
    it('StepsBar click event', () => {
        stepsBar.find(`.${styles.stepsBar}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
