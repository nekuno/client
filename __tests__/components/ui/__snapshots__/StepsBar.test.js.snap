// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test StepsBar component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StepsBar
    canContinue={true}
    cantContinueText="You cannot continue"
    continueText="Continue"
    currentStep={0}
    onClickHandler={[MockFunction]}
    totalSteps={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="stepsBar undefined"
        onClick={[Function]}
      >
        <ul
          className="dots"
        >
          <li
            className="active"
          >
            <div
              className="dot"
            />
          </li>
          <li
            className={null}
          >
            <div
              className="dot"
            />
          </li>
          <li
            className={null}
          >
            <div
              className="dot"
            />
          </li>
        </ul>
        <div
          className="continue small"
        >
          Continue 
          <span
            className="icon icon-arrow-right"
          />
        </div>
      </div>,
      "className": "stepsBarWrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ul
            className="dots"
          >
            <li
              className="active"
            >
              <div
                className="dot"
              />
            </li>
            <li
              className={null}
            >
              <div
                className="dot"
              />
            </li>
            <li
              className={null}
            >
              <div
                className="dot"
              />
            </li>
          </ul>,
          <div
            className="continue small"
          >
            Continue 
            <span
              className="icon icon-arrow-right"
            />
          </div>,
        ],
        "className": "stepsBar undefined",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="active"
              >
                <div
                  className="dot"
                />
              </li>,
              <li
                className={null}
              >
                <div
                  className="dot"
                />
              </li>,
              <li
                className={null}
              >
                <div
                  className="dot"
                />
              </li>,
            ],
            "className": "dots",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="dot"
                />,
                "className": "active",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "dot",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="dot"
                />,
                "className": null,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "dot",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="dot"
                />,
                "className": null,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "dot",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Continue ",
              <span
                className="icon icon-arrow-right"
              />,
            ],
            "className": "continue small",
          },
          "ref": null,
          "rendered": Array [
            "Continue ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "icon icon-arrow-right",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="stepsBar undefined"
          onClick={[Function]}
        >
          <ul
            className="dots"
          >
            <li
              className="active"
            >
              <div
                className="dot"
              />
            </li>
            <li
              className={null}
            >
              <div
                className="dot"
              />
            </li>
            <li
              className={null}
            >
              <div
                className="dot"
              />
            </li>
          </ul>
          <div
            className="continue small"
          >
            Continue 
            <span
              className="icon icon-arrow-right"
            />
          </div>
        </div>,
        "className": "stepsBarWrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ul
              className="dots"
            >
              <li
                className="active"
              >
                <div
                  className="dot"
                />
              </li>
              <li
                className={null}
              >
                <div
                  className="dot"
                />
              </li>
              <li
                className={null}
              >
                <div
                  className="dot"
                />
              </li>
            </ul>,
            <div
              className="continue small"
            >
              Continue 
              <span
                className="icon icon-arrow-right"
              />
            </div>,
          ],
          "className": "stepsBar undefined",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="active"
                >
                  <div
                    className="dot"
                  />
                </li>,
                <li
                  className={null}
                >
                  <div
                    className="dot"
                  />
                </li>,
                <li
                  className={null}
                >
                  <div
                    className="dot"
                  />
                </li>,
              ],
              "className": "dots",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="dot"
                  />,
                  "className": "active",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "dot",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="dot"
                  />,
                  "className": null,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "dot",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="dot"
                  />,
                  "className": null,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "dot",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Continue ",
                <span
                  className="icon icon-arrow-right"
                />,
              ],
              "className": "continue small",
            },
            "ref": null,
            "rendered": Array [
              "Continue ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon icon-arrow-right",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
